include/glow/Base/Tensor.h

```diff
@@ -82,6 +123,33 @@ class Tensor final {
   char *getData() const { return data_; }
 
 public:
+  /// \returns true if tensor data is stored on a device
+  bool isDeviceResident() const { return residencyInfoP_->isDeviceResident(); }
+
+  // Update device residency info with new device manager and context
+  void moveToDevice(runtime::DeviceManager *deviceManager, void *context) {
```

Comment:
Question for a future diff: To support pinned Tensors, should we have an optional argument here to control whether tensorResidency_ is Device or Host with DeviceManager state?

Reply:
What is the use case for pinned tensors - 
Do want to be able to copy from pinned memory directly to device and vice-versa? (meaning: do we need 2 pointers (to opaque memory) in the tensor class for both pinned location and device location?)

Reply:
Yes we want to copy into pinned memory ahead of time and then move it to the device from that memory. I think we can store the extra pointer in the DeviceResidencyInfo for that backend? Unsure, but we dont need to solve it for this diff.